packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -1591,8 +1626,49 @@ export class CacheDataClient implements IDataClient {
     }
 
     return await this.rateLimited(async () => {
-      const itemsToUse = this.convertSetBatchElements(items);
-      const ttlToUse = ttl || this.defaultTtlSeconds;
+      let itemsToUse: [Uint8Array, Uint8Array][] =
+        this.convertSetBatchElements(items);
+      if (options?.compress) {
+        this.logger.trace(
+          'CacheClient.setBatch; compression enabled, calling value compressor'
+        );
+        if (this.valueCompressor === undefined) {
+          return this.cacheServiceErrorMapper.returnOrThrowError(
+            new CompressionError('CacheClient.setBatch', 'compress'),
+            err => new CacheSetBatch.Error(err)
+          );
+        }
+        const compressor: ICompression = this.valueCompressor;
+        itemsToUse = await Promise.all(
+          itemsToUse.map(async ([key, value]) => {
+            return [
+              key,
+              await compressor.compress(
+                this.configuration.getCompressionStrategy()?.compressionLevel ??
+                  CompressionLevel.Balanced,
```

Comment:
We've duplicated this in a few places now. Let's set this to an instance variable on initialization since the compression strategy is read only hence will not change.

Reply:
sure